drivers/pppos.c

```diff
@@ -25,7 +25,52 @@
 #include <sys/msg.h>
 #include <syslog.h>
 
+/* Intentionally does not begin from 1. */
+#define PPPOS_MODEM_GENERIC 2
```

Comment:
Note: this was actually meant for a Telit LTE modem, not a generic one, but I don't know if we want to specify it that much (it may work with other modems also).

Note for the future (I'm okay with the current approach, but it's not extensible in regards of adding many more modems):
I would prefer to put the configuration in separate headers (one modem per header, eg. `pppos_modem.h` in subdirs eg. `modem/[modem_name]/pppos_modem.h`) and just decide on the modem in Makefile (add correct include path to `CFLAGS`). This would make adding new modems less painful.

@anglov, do You think these per-modem defines are enough to provide support for LTE cat M1 modem You were experimenting with without too many additional changes?

Reply:
In cat M1 modem there was one issue: when modem go into data mode, they doesn't back to AT mode after '+++'. Maybe just some additional configuration was needed but I didn't find solution for that.

In fact I just need to disable code for setting APN (chosing the default one), disablr action that needs to back to AT mode and change init commands.

So, it's possible that different at_disconnect will be needed for cat M1, but not sure - I don't find proper solution  for that modem.

What is missing and could be problematic in future is set non-default APN if configuration file is not used. Probably not an issue for this PR.

Reply:
Maybe until it is known to be generic (tested), let's call it `telit`. Regarding separate headers, at first I thought it would unnecessarily spawn many tiny files. I have separated them anyway and it looks much more manageable now - only a new subdirectory and a header are needed to support a new modem (no changes to phoenix-rtos-lwip needed), and a `PPPOS_MODEM` define set appropriately in a project.

Reply:
I like the current version :). I think there should be no default - If You want to use PPPoS driver in the project, You should also explicitly specify the modem You would like to support.